### ISSUE TITLE
Program loops indefinitely when I start an application with plot set on autosampling.

### ISSUE THREAD

**Issue by Mriv31** [pyqtgraph/pyqtgraph#2713]
### Short description
When I start a pyqtgraph application with a plot with auto Downsampling, it loops over different values of the downsampling parameter and then crashes. I also have remarks about the optimization of the Downsampling but will put them in a discussion.

### Code to reproduce

```
import pyqtgraph as pg
from PyQt6 import QtCore, QtWidgets

import numpy as np
pp = pg.mkQApp()
mw = QtWidgets.QMainWindow()
mw.setWindowTitle('pyqtgraph example: PlotWidget')
mw.resize(800,800)
cw = QtWidgets.QWidget()
mw.setCentralWidget(cw)
l = QtWidgets.QVBoxLayout()
cw.setLayout(l)

pw = pg.PlotWidget()  ## giving the plots names allows us to link their axes together
pw.setClipToView( True)
pw.enableAutoRange(False)
pw.setDownsampling( ds=None, auto=1, mode="subsample")
l.addWidget(pw)
data = np.arange(10000000)+np.random.normal(0,10,10000000)
pw.plot(data, title="Simplest possible plotting example")
mw.show()

if __name__ == '__main__':
    pg.exec()
```

### Real behavior
The program loops. When I print the "ds" parameter, it goes down until the program crashes.

### Tested environment(s)

 * PyQtGraph version: 0.13.3 
 * Qt Python binding: PyQt6
 * Python version: Python 3.10
 * NumPy version: 1.21.5
 * Operating system: Ubuntu
 * Installation method: pip

**Comment by pijyoi** [comment]
I can't reproduce the crash. It seems to run fine for me. Windows 11 / Python 3.10 / PyQt6 6.5.0 / numpy 1.24.3

Simplified version:
```python
import pyqtgraph as pg
import numpy as np

pg.mkQApp()
pw = pg.PlotWidget()
pw.setClipToView(True)
pw.enableAutoRange(False)
pw.setDownsampling(ds=None, auto=1, mode="subsample")
data = np.arange(10000000)+np.random.normal(0,10,10000000)
pw.plot(data, title="Simplest possible plotting example")
pw.show()
pg.exec()
```

**Comment by Mriv31** [comment]
Maybe try to resize the window. It doesn't happen for all sizes. I tested your code and it does the same, whether in 0.13.3 or 0.13.4-dev0 (still ubuntu). If I add ``` print("ds = ", ds)``` just  before ```if self.opts['clipToView']:``` in ```_getDisplayDataset```, I get  the following infinite loop : 
```ds =  1
ds =  1
ds =  1
ds =  1
ds =  3595
ds =  3595
ds =  2886
ds =  2936
ds =  2935
ds =  2936
ds =  2935
ds =  2936
ds =  2935
ds =  2936
...
```

I found a quick fix which is to only allow  discrete logarithmically spaced values for ds as explained here : https://github.com/pyqtgraph/pyqtgraph/discussions/2710. That avoids that loop between the automatic computation of ```ds``` and the range update.

Btw the fact that ```ds = 1``` at first is also a problem. If you have a large dataset you would like it to be auto downsampled from the beginning, if not the first  plot update will crash the program. The problem comes from the fact that the ```PlotDataItem``` gets updated before  the autodownsample or downsample properties get passed to it by the ```PlotItem```. Also gave a quick fix there https://github.com/pyqtgraph/pyqtgraph/discussions/2710 , in the section "PlotDataItem update problem bypass. I just pass the autodownsample argument to the ```PlotDataItem``` at the creation even if they are seat for the ```PlotItem```. It must be possible to make that better.

Best
